### ISSUE TITLE
[Bug]: v9 non-modal dialog does not close on document escape

### ISSUE THREAD

**Issue by YuanboXue-Amber** [microsoft/fluentui#26918]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
v9.15.5
```

### Are you reporting Accessibility issue?

Yes

### Reproduction

https://react.fluentui.dev/?path=/docs/components-dialog--default#no-focusable-element

## Actual Behavior
1. Open: https://react.fluentui.dev/?path=/docs/components-dialog--default#no-focusable-element and then Open this example in codesandbox, which brings us to: https://codesandbox.io/s/gbc5ps?file=/example.tsx
2. Open the codesandbox example in full screen to make sure there's no iframe.
3. Click on "Open non-modal dialog", a dialog without backdrop opens
4. Tabbing to focus on any of the two trigger button in the page, press ESC, the dialog stays open

## Expected Behavior
Dialog should close on step 4

### Requested priority

Low

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by Hotell** [comment]
This behaviour actually happens on both of these examples. Once the focus is lost from the dialog, one can never re-foocus again (via tab, or mouse) this the ESC keypress won't be catched.

this might be limitation tabster limitation ? if its an expected behaviour docs should be updated to not confuse users.

**Comment by ozluy** [comment]
![Screenshot 2023-03-15 at 23 44 54](https://user-images.githubusercontent.com/2522722/225460061-4086e6d7-a79b-44d9-8d8f-466beca32d39.png)
I guess this message says it all,

By the way if you focus out it does not work but if focus in (click the content of dialog) without opening another modal it works.

**Comment by bsunderhus** [comment]
This is a scenario that is declared as a `don't` on our best practices for the `Dialog`.
<img width="500" alt="image" src="https://user-images.githubusercontent.com/5483269/226427388-260fcadc-605c-4bd9-9a62-4ab0990827af.png">
We do not support having a dialog without a focusable element!  Therefore this issue is expected behaviour!

We should not care much for Escape keyboard support for a case we do not support (it would require extra Kb of code for something we actually don't want our costumers doing!). 

That being said supporting escape keydown to close a modal without focusable element can be easily implemented on user land, by just adding an event listener to `window`


```tsx

export const NoFocusableElement = () => {
  const [open, setOpen] = React.useState(false);
  React.useEffect(() => {
    if (open) {
      const handleKeyDown = (event: KeyboardEvent) => {
        if (event.key === Escape && open) {
          setOpen(false);
        }
      };
      window.addEventListener('keydown', handleKeyDown);
      return () => window.removeEventListener('keydown', handleKeyDown);
    }
  }, [open]);
  return (
    <>
      <Dialog open={open} onOpenChange={(_, data) => setOpen(data.open)} modalType="non-modal">
        <DialogTrigger disableButtonEnhancement>
          <Button>Open non-modal dialog</Button>
        </DialogTrigger>
        <DialogSurface>
          <DialogBody>
            <DialogTitle action={null}>Dialog Title</DialogTitle>
            <DialogContent>
              <p>⛔️ A modal Dialog without focusable elements is not recommended!</p>
              <p>✅ Escape key works</p>
            </DialogContent>
          </DialogBody>
        </DialogSurface>
      </Dialog>
    </>
  );
};

```
